bug: action_index test_malformed_records_counted_and_backfill_processes_two_days is a 7-day calendar time bomb

## What's broken

\`tests/test_action_index.py::test_malformed_records_counted_and_backfill_processes_two_days\` hardcodes prompt-day fixture dates \`2026-08-24\`/\`2026-08-25\`. \`build_action_index\` calls \`_rotate_and_prune(index_dir, today)\` at the end of every run, which archives (gzips) any index day-file older than \`_ARCHIVE_AFTER_DAYS = 7\` calendar days from the **real** wall-clock date (\`datetime.now(timezone.utc)\`), not a fixture-relative date.

As of 2026-08-31 (exactly 7 days after 2026-08-24), the test's own \`2026-08-24.jsonl\` output file gets archived to \`.jsonl.gz\` mid-test, before the test's final assertion checks for a plain \`.jsonl\` file — causing a deterministic failure that will now recur every day going forward as the hardcoded dates fall further behind the real calendar, and is currently failing CI on \`main\` itself (independent of any feature PR):

\`\`\`
AssertionError: assert {'2026-08-25.jsonl'} == {'2026-08-24.jsonl', '2026-08-25.jsonl'}
\`\`\`

## Impact

- Blocks CI (\`test (3.11/3.12/3.13)\`) on \`main\` and on every open PR, unrelated to their actual changes.
- Discovered while delivering #1118 (verdict/expected_outcome work), which is otherwise unrelated — filed separately per scope-isolation rules.

## Scope

- Fix `tests/test_action_index.py`'s fixture to be robust to the real calendar (e.g. derive day names from `datetime.now(timezone.utc)` relative offsets instead of hardcoded absolute dates), OR inject/monkeypatch the "today" used by `_rotate_and_prune` in this test.
- Do not change `_ARCHIVE_AFTER_DAYS`/`_DEFAULT_RETENTION_DAYS` production behavior — this is a test-fixture bug, not a feature change.

## Acceptance criteria

- The test passes today (2026-08-31) and will keep passing on any future date (no more hardcoded absolute dates as fixture inputs, or the archive-cutoff arithmetic used by the test is pinned to the same relative reference as the fixture).
- Full test suite green; CI green on all 3 Python versions.

## Out of scope

- No behavior change to `action_index.py` itself.

--- a/tests/test_action_index.py
+++ b/tests/test_action_index.py
@@ -2,9 +2,33 @@
 from __future__ import annotations
 
 import json
+from datetime import datetime, timezone
 from pathlib import Path
 
+import pytest
+
+from nanobot.runtime import action_index
 from nanobot.runtime.action_index import build_action_index, normalize_action
+
+
+# #1124: every fixture in this file hardcodes prompt-day dates in the
+# 2026-08-23..2026-08-25 window. build_action_index()'s own call to
+# _rotate_and_prune() at the end of every run reads the REAL wall-clock date
+# via datetime.now(timezone.utc) to decide which day-files are >=7 days old
+# and should be archived (gzipped) — a calendar time bomb, since the fixture
+# dates fall further behind the real date every day this suite exists.
+# Freeze "now" to a date inside the fixtures' own window so the archive/
+# retention math this module performs is stable regardless of when the test
+# actually runs, without changing any production behavior.
+class _FrozenDatetime(datetime):
+    @classmethod
+    def now(cls, tz=None):
+        return cls(2026, 8, 25, 12, 0, 0, tzinfo=tz or timezone.utc)
+
+
+@pytest.fixture(autouse=True)
+def _frozen_calendar(monkeypatch):
+    monkeypatch.setattr(action_index, "datetime", _FrozenDatetime)
 
 
 def _write(path: Path, rows: list[dict] | list[str]) -> None:

